ui/apps/platform/src/Components/CompoundSearchFilter/components/SearchFilterAutocomplete.tsx

```diff
@@ -0,0 +1,288 @@
+import React, { useMemo, useRef, useState } from 'react';
+import {
+    Select,
+    SelectOption,
+    SelectList,
+    SelectOptionProps,
+    MenuToggle,
+    MenuToggleElement,
+    TextInputGroup,
+    TextInputGroupMain,
+    TextInputGroupUtilities,
+    Button,
+    Skeleton,
+    Flex,
+    debounce,
+} from '@patternfly/react-core';
+import TimesIcon from '@patternfly/react-icons/dist/esm/icons/times-icon';
+import { ArrowRightIcon } from '@patternfly/react-icons';
+import { useQuery } from '@apollo/client';
+import SEARCH_AUTOCOMPLETE_QUERY, {
+    SearchAutocompleteQueryResponse,
+} from 'queries/searchAutocomplete';
+
+type SearchFilterAutocompleteProps = {
+    searchCategory: string;
+    searchTerm: string;
+    value: string;
+    onChange: (value: string) => void;
+    onSearch: (value: string) => void;
+    textLabel: string;
+};
+
+function getSelectOptions(
+    data: SearchAutocompleteQueryResponse | undefined,
+    isLoading: boolean
+): SelectOptionProps[] {
+    if (isLoading) {
+        return [
+            {
+                isDisabled: true,
+                value: 'autocomplete-options-loading',
+                children: (
+                    <Flex
+                        direction={{ default: 'column' }}
+                        spaceItems={{ default: 'spaceItemsMd' }}
+                    >
+                        <Skeleton screenreaderText="Loading suggested options 1" width="100%" />
+                        <Skeleton screenreaderText="Loading suggested options 2" width="75%" />
+                        <Skeleton screenreaderText="Loading suggested options 3" width="90%" />
+                    </Flex>
+                ),
+            },
+        ];
+    }
+
+    if (data && data.searchAutocomplete && data.searchAutocomplete.length !== 0) {
+        const options: SelectOptionProps[] = data.searchAutocomplete.map((optionValue) => {
+            return {
+                value: optionValue,
+                children: optionValue,
+            };
+        });
+        return options;
+    }
+
+    return [
+        {
+            isDisabled: false,
+            children: `No results found`,
+            value: 'no results',
+        },
```

Comment:
There is a quirk with the existing autocomplete that sometimes shows the "No results found" text and sometimes shows the "Add <text>" option, but I think all we really need is the latter. This might be good to discuss with UX, but I think the following would be ideal behavior:

1. If the user is entering text, during the debounce window or while loading results, we should show both the skeleton loaders and the "Add <text>" option. In this case, the "Add" option should be highlighted as the active item and pressing "Enter" will apply the user's text as a filter immediately.
2. When the debounce is complete, if there are results returned, the "Add <text>" option should be last in the list and **not** the active item. _Should pressing "Enter" at this point do anything at all? If so, it should be clear which item in the list is "active" and will be applied as a filter._
3. When the debounce is complete, if there are no results returned, the "Add <text>" option should be the only item and pressing "Enter" should apply it as a filter once again.

Reply:
Yeah, this is a good topic for us to discuss in the next meeting 👀